File: CountLines/count_lines_1.py
#!/bin/env python
# -*- coding: utf-8 -*-

import sys

def count_lines( file_path ) :
  counter = 0

  with open( file_path, 'r' ) as file :
    for line in file :
      counter += 1

  return counter

if __name__ == '__main__' :

  import doctest
  doctest.testmod()

  print count_lines( sys.argv[ 1 ] )

File: CountLines/count_lines_1.pl
#!/bin/env perl

use strict;
use warnings;

# http://learn.perl.org/faq/perlfaq5.html#How-do-I-count-the-number-of-lines-in-a-file
sub count_lines {
  my $filename = shift;

  open( FILE, $filename );
  while ( <FILE> ) {
  }
  my $lines = $.;
  close FILE;

  return $lines;
}

if ( ! caller ) {
  print count_lines( $ARGV[ 0 ] );
}
